[Bug] [connector] [ftp] Unsupported options found for 'ftp-x'

### Search before asking

- [X] I had searched in the [issues](https://github.com/DTStack/chunjun/issues) and found no similar issues.


### What happened

当从chunjun flink 1.12 ftp->ftp sql 迁移到  1.16 版本后 ，


```
Caused by: org.apache.flink.table.api.ValidationException: Unsupported options found for 'ftp-x'.

Unsupported options:

compress-type
connect-pattern
field-delimiter
file-extension
first-line-header
port

Supported options:

connector
encoding
file-name
fileType
format
host
isFirstLineHeader
max-file-size
password
path
property-version
protocol
timeout
username
	at org.apache.flink.table.factories.FactoryUtil.validateUnconsumedKeys(FactoryUtil.java:631)
	at org.apache.flink.table.factories.FactoryUtil$FactoryHelper.validate(FactoryUtil.java:923)
	at com.dtstack.chunjun.connector.ftp.table.FtpDynamicTableFactory.createDynamicTableSource(FtpDynamicTableFactory.java:91)
	at org.apache.flink.table.factories.FactoryUtil.createDynamicTableSource(FactoryUtil.java:163)
	... 48 more
```

### What you expected to happen

--

### How to reproduce


```
CREATE TABLE SOURCE (
    C_ID INT,
    C_BOOLEAN BOOLEAN,
    C_TINYINT TINYINT,
    C_SMALLINT SMALLINT,
    C_INT INT,
    C_BIGINT BIGINT,
    C_FLOAT FLOAT,
    C_DOUBLE DOUBLE,
    C_DECIMAL DECIMAL,
    C_STRING STRING,
    C_VARCHAR VARCHAR(255),
    C_CHAR CHAR(255),
    C_TIMESTAMP TIMESTAMP,
    C_DATE DATE
) WITH (
    'connector' = 'ftp-x',
    -- 路径不支持通配符*,支持传目录或者具体文件地址
    'path' = '/flinkx-test/source-zip',
    -- 服务器访问协议，目前支持ftp、sftp
    'protocol' = 'ftp',
    'host' = 'xxxx',
    -- 默认值：若传输协议是sftp协议，默认值是22；若传输协议是标准ftp协议，默认值是21
    'port' = '21',
    'username' = 'xxxx',
    'password' = 'xxx',
    -- 文件的类型，和原生flink保持一致，支持原生所有类型。 读取的文件类型，默认取文件后缀名，支持CSV,TXT,EXCEL
    'format' = 'csv',
    -- protocol为ftp时的连接模式，可选PASV和PORT
    'connect-pattern' = 'PASV',
    -- 连接超时时间，单位毫秒
    'timeout' = '5000',
    -- isFirstLineHeader 首行是否为标题行，如果是则不读取第一行 默认值：false
    'first-line-header' = 'true',
    -- 读取的文件类型，默认取文件后缀名，支持CSV,TXT,EXCEL
    'file-extension' = 'TXT',
    -- 读取的字段分隔符 默认值：,
    'field-delimiter' = '\t',
    -- 读取文件的编码配置 默认值：UTF-8
    'encoding' = 'UTF-8',
    -- 文件压缩类型,支持ZIP 默认值：无
    'compress-type' = 'ZIP'
    -- 以下sql模式暂不支持的参数
    -- fileConfig 文件参数配置 csv文件是否进行trim：`"fileConfig":{"trimWhitespace":true}`
);

CREATE TABLE SINK (
    C_ID INT,
    C_BOOLEAN BOOLEAN,
    C_TINYINT TINYINT,
    C_SMALLINT SMALLINT,
    C_INT INT,
    C_BIGINT BIGINT,
    C_FLOAT FLOAT,
    C_DOUBLE DOUBLE,
    C_DECIMAL DECIMAL,
    C_STRING STRING,
    C_VARCHAR VARCHAR(255),
    C_CHAR CHAR(255),
    C_TIMESTAMP TIMESTAMP,
    C_DATE DATE
) WITH (
    'connector' = 'ftp-x',
    -- 会自动创建目录
    'path' = '/flinkx-test/sink',
    -- 服务器访问协议，目前支持ftp、sftp
    'protocol' = 'ftp',
    'host' = 'xxx',
    'port' = '21',
    'username' = 'xxx',
    'password' = 'xxx',
    -- 文件的类型，和原生flink保持一致，支持原生所有类型 默认值：csv
    'format' = 'csv',
    -- 文件名, 当文件大小超过1G时，会此名_x_x 命名文件
    'file-name' = 'flinkx-sink',
    -- 文件扩展名
    'file-extension' = '.log',
    -- 写入的的字段分隔符 默认值：,
    'csv.field-delimiter' = '\t'
    -- 不支持 输出标题行
    -- 写入ftp单个文件最大大小，单位字节, 默认大小1G
    ,'max-file-size' = '10485760' -- 10M
);


INSERT INTO SINK
SELECT *
FROM SOURCE;



```

### Anything else

_No response_

### Version

master

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

--- a/chunjun-connectors/chunjun-connector-ftp/src/main/java/com/dtstack/chunjun/connector/ftp/options/FtpOptions.java
+++ b/chunjun-connectors/chunjun-connector-ftp/src/main/java/com/dtstack/chunjun/connector/ftp/options/FtpOptions.java
@@ -73,15 +73,27 @@ public class FtpOptions extends BaseFileOptions {
                     .defaultValue("PASV")
                     .withDescription("ftp connectPattern");
 
-    public static final ConfigOption<Boolean> isFirstLineHeader =
-            ConfigOptions.key("isFirstLineHeader")
+    public static final ConfigOption<Boolean> FIRST_LINE_HEADER =
+            ConfigOptions.key("first-line-header")
                     .booleanType()
                     .defaultValue(true)
-                    .withDescription("FirstLineHeader");
+                    .withDescription("skip header");
 
     public static final ConfigOption<String> FILE_TYPE =
-            ConfigOptions.key("fileType")
+            ConfigOptions.key("file-type")
                     .stringType()
                     .defaultValue("TXT")
-                    .withDescription("FILE_TYPE");
+                    .withDescription("file type");
+
+    public static final ConfigOption<String> FIELD_DELIMITER =
+            ConfigOptions.key("field-delimiter")
+                    .stringType()
+                    .defaultValue(",")
+                    .withDescription("field delimiter");
+
+    public static final ConfigOption<String> COMPRESS_TYPE =
+            ConfigOptions.key("compress-type")
+                    .stringType()
+                    .noDefaultValue()
+                    .withDescription("compress type");
 }

--- a/chunjun-connectors/chunjun-connector-ftp/src/main/java/com/dtstack/chunjun/connector/ftp/table/FtpDynamicTableFactory.java
+++ b/chunjun-connectors/chunjun-connector-ftp/src/main/java/com/dtstack/chunjun/connector/ftp/table/FtpDynamicTableFactory.java
@@ -20,10 +20,13 @@ package com.dtstack.chunjun.connector.ftp.table;
 
 import com.dtstack.chunjun.config.FieldConfig;
 import com.dtstack.chunjun.config.TypeConfig;
+import com.dtstack.chunjun.connector.ftp.config.ConfigConstants;
 import com.dtstack.chunjun.connector.ftp.config.FtpConfig;
 import com.dtstack.chunjun.connector.ftp.options.FtpOptions;
 import com.dtstack.chunjun.connector.ftp.sink.FtpDynamicTableSink;
 import com.dtstack.chunjun.connector.ftp.source.FtpDynamicTableSource;
+import com.dtstack.chunjun.table.options.BaseFileOptions;
+import com.dtstack.chunjun.util.StringUtil;
 
 import org.apache.flink.api.common.serialization.DeserializationSchema;
 import org.apache.flink.api.common.serialization.SerializationSchema;
@@ -60,7 +63,18 @@ public class FtpDynamicTableFactory implements DynamicTableSourceFactory, Dynami
         ftpConfig.setPassword(config.get(FtpOptions.PASSWORD));
 
         ftpConfig.setEncoding(config.get(FtpOptions.ENCODING));
+        ftpConfig.setFirstLineHeader(config.get(FtpOptions.FIRST_LINE_HEADER));
+        ftpConfig.setFtpFileName(config.get(FtpOptions.FILE_NAME));
+        ftpConfig.setMaxFileSize(config.get(BaseFileOptions.MAX_FILE_SIZE));
+        ftpConfig.setCompressType(config.get(FtpOptions.COMPRESS_TYPE));
         ftpConfig.setFileType(config.get(FtpOptions.FILE_TYPE));
+
+        if (!ConfigConstants.DEFAULT_FIELD_DELIMITER.equals(
+                config.get(FtpOptions.FIELD_DELIMITER))) {
+            String fieldDelimiter =
+                    StringUtil.convertRegularExpr(config.get(FtpOptions.FIELD_DELIMITER));
+            ftpConfig.setFieldDelimiter(fieldDelimiter);
+        }
 
         if (config.get(FtpOptions.TIMEOUT) != null) {
             ftpConfig.setTimeout(config.get(FtpOptions.TIMEOUT));
@@ -76,8 +90,8 @@ public class FtpDynamicTableFactory implements DynamicTableSourceFactory, Dynami
             ftpConfig.setPort(config.get(FtpOptions.PORT));
         }
 
-        if (config.get(FtpOptions.isFirstLineHeader) != null) {
-            ftpConfig.setFirstLineHeader(config.get(FtpOptions.isFirstLineHeader));
+        if (config.get(FtpOptions.FIRST_LINE_HEADER) != null) {
+            ftpConfig.setFirstLineHeader(config.get(FtpOptions.FIRST_LINE_HEADER));
         }
         return ftpConfig;
     }
@@ -117,7 +131,7 @@ public class FtpDynamicTableFactory implements DynamicTableSourceFactory, Dynami
         final FactoryUtil.TableFactoryHelper helper =
                 FactoryUtil.createTableFactoryHelper(this, context);
         final ReadableConfig config = helper.getOptions();
-        helper.validate();
+        helper.validateExcept("csv.");
 
         ResolvedSchema resolvedSchema = context.getCatalogTable().getResolvedSchema();
 
@@ -159,8 +173,12 @@ public class FtpDynamicTableFactory implements DynamicTableSourceFactory, Dynami
         options.add(FtpOptions.ENCODING);
         options.add(FtpOptions.MAX_FILE_SIZE);
         options.add(FtpOptions.FORMAT);
-        options.add(FtpOptions.isFirstLineHeader);
+        options.add(FtpOptions.PORT);
         options.add(FtpOptions.FILE_TYPE);
+        options.add(FtpOptions.CONNECT_PATTERN);
+        options.add(FtpOptions.FIRST_LINE_HEADER);
+        options.add(FtpOptions.FIELD_DELIMITER);
+        options.add(FtpOptions.COMPRESS_TYPE);
         return options;
     }
 }

